tests/unit/components/panel/AttributeSection.spec.js

```diff
@@ -0,0 +1,325 @@
+            type: 'Object',
+            value: '',
+          },
+          isRoot: true,
+        },
+      });
+      expect(wrapper.vm.isObject).toEqual(true);
+    });
+  });
+
+  describe('Test watcher: props.attribute', () => {
+    it('should update localAttribute', async () => {
+      expect(wrapper.vm.localAttribute).toEqual({
+        name: 'attribute',
+        type: 'String',
+        value: [],
+      });
+      expect(wrapper.vm.isObject).toEqual(false);
+
+      await wrapper.setProps({
+        attribute: {
+          name: 'test',
+          type: 'Object',
+        },
+      });
+      expect(wrapper.vm.localAttribute).toEqual({
+        name: 'test',
+        type: 'Object',
+      });
+    });
+
+    it('should update isObject if new attribute is not an object', async () => {
+      await wrapper.setProps({
+        attribute: {
+          name: 'test',
+          type: 'Object',
+        },
+      });
+      expect(wrapper.vm.isObject).toEqual(true);
+
+      await wrapper.setProps({
+        attribute: {
+          name: 'test',
+          type: 'String',
+        },
+      });
+      expect(wrapper.vm.isObject).toEqual(false);
+    });
+
+    it('should update isObject if new attribute is an object', async () => {
+      expect(wrapper.vm.localAttribute).toEqual({
+        name: 'attribute',
+        type: 'String',
+        value: [],
+      });
+      expect(wrapper.vm.isObject).toEqual(false);
+
+      await wrapper.setProps({
+        attribute: {
+          name: 'test',
+          type: 'Object',
+        },
+      });
+      expect(wrapper.vm.isObject).toEqual(true);
+    });
+  });
+
+  describe('Test function: getSubAttributes', () => {
+    it('should return empty attributes array on attribute without definition and empty value', () => {
+      expect(wrapper.vm.getSubAttributes({
+        value: [],
+        definition: null,
+      })).toEqual([]);
+    });
+
+    it('should return instantiate defined attribute with its value even if value is empty', () => {
+      expect(wrapper.vm.getSubAttributes({
+        value: [],
+        definition: {
+          definedAttributes: [{
+            name: 'test',
+            type: 'String',
+          }],
+        },
+      })).toEqual([new ComponentAttribute({
+        name: 'test',
+        type: 'String',
+        definition: {
+          name: 'test',
+          type: 'String',
+        },
+      })]);
+    });
+
+    it('should return instantiate defined attribute with its value even if value is not empty', () => {
+      expect(wrapper.vm.getSubAttributes({
+        value: [{
+          name: 'test',
+          value: 'toto',
+          type: 'String',
+          definition: {
+            name: 'test',
+            type: 'String',
+          },
+        }],
+        definition: {
+          definedAttributes: [{
+            name: 'test',
+            type: 'String',
+          }],
+        },
+      })).toEqual([new ComponentAttribute({
```

Comment:
define as variable and use it in getSubAttributes too

Reply:
done